webapp/src/main/java/com/box/l10n/mojito/service/cache/DatabaseCache.java

```diff
@@ -0,0 +1,226 @@
+            // target class structure).
+            try {
+                Object deserializedValue = deserializeCacheValue(bytes);
+                return this.fromStoreValue(deserializedValue);
+            } catch (Exception e) {
+                logger.error(String.format("Could not deserialize cache entry. Problematic serialized value: %s . Exception: ", Arrays.toString(bytes)), e);
+
+                if (cacheConfig.isEvictEntryOnDeserializationFailure()) {
+                    logger.error(String.format("The entry failing deserialization will be evicted permanently. Problematic serialized value: %s . Exception: ", Arrays.toString(bytes)), e);
+                    evict(key);
+                    return null;
+                }
+
+                throw e;
+            }
+        }
+
+        return null;
+    }
+
+    @Override
+    public String getName() {
+        return this.cacheName;
+    }
+
+    @Override
+    public Object getNativeCache() {
+        return applicationCacheRepository;
+    }
+
+    @Override
+    public <T> T get(Object key, Callable<T> valueLoader) {
+        Object cacheEntry = lookup(key);
+        if (cacheEntry != null) {
+            //noinspection unchecked
+            return (T) cacheEntry;
+        }
+
+        try {
+            T computedValue = valueLoader.call();
+            this.put(key, serializeCacheValue(computedValue));
+            return computedValue;
+        } catch (Throwable ex) {
+            throw new ValueRetrievalException(key, valueLoader, ex);
+        }
+    }
+
+    @Transactional
+    @Override
+    public void put(Object key, Object value) {
+        checkCacheTypeConfigured();
+
+        long ttlInSeconds = cacheConfig.getTtl().get(ChronoUnit.SECONDS);
+        if (ttlInSeconds <= 0) {
+            applicationCacheRepository.upsertNoExpiryDate(
+                    cacheType.getId(),
+                    getCacheKeyMD5(key),
+                    serializeCacheValue(value));
+        } else {
+            applicationCacheRepository.upsertWithTTL(
+                    cacheType.getId(),
+                    getCacheKeyMD5(key),
+                    serializeCacheValue(value),
+                    ttlInSeconds);
+        }
+
+        scheduleProbabilisticExpiredEvictionAsync();
+    }
+
+    @Transactional
+    @Override
+    public void evict(Object key) {
+        checkCacheTypeConfigured();
+        applicationCacheRepository.deleteById(getApplicationCacheId(key));
+    }
+
+    @Transactional
+    @Override
+    public void clear() {
+        checkCacheTypeConfigured();
+        applicationCacheRepository.clearCache(cacheType.getId());
+    }
+
+    @Pollable(async = true, message = "Cleaning up expired cache entries.")
+    private void scheduleProbabilisticExpiredEvictionAsync() {
+        if (rand.nextDouble() < cacheConfig.getWritesToExpiredEvictionRatio()) {
```

Comment:
the "WritesToExpiredEvictionRatio" concept is not obvious to me at first glance.

Reply:
i don't have strong feeling on this one but I think we'd fine with a cronjob for now. We do have similar job already to perform cleanups